server/services/DeckService.js

```diff
@@ -84,9 +110,20 @@ class DeckService {
         return this.decks.update({ _id: deck.id }, { '$set': properties });
     }
 
-    delete(id) {
+    async delete(id) {
+        let previousVersion = await this.getById(id);
+        //do not delete the deck if the deck is locked
+        if(previousVersion.locked) {
+            return () => Promise.resolve();
+        }
         return this.decks.remove({ _id: id });
     }
+
+    async userAlreadyHasDeckForEvent(userName, eventId) {
+        let otherDecksForThisUser = await this.findByUserName(userName);
```

Comment:
This should operate on the decks collection directly and filter at source, otherwise you're returning far more records than necessary and the work can be done on the db

Reply:
good idea, done